src/contrib/transports/Akka.Remote.Transport.Streams.Tcp/Class1.cs

```diff
@@ -0,0 +1,313 @@
+                    _listenAddress =
+                        new DnsEndPoint(Settings.Hostname, Settings.Port);
+                    _outAddr = new DnsEndPoint(Settings.Hostname,0);
+                }
+            }
+            
+            
+            _connectionSource =
+                System.TcpStream().Bind(Settings.Hostname, Settings.Port,
+                    options: ImmutableList.Create<Inet.SocketOption>(
+                        new Inet.SO.SendBufferSize(Settings.SendBufferSize
+                                                   ?? 256 * 1024),
+                        new Inet.SO.ReceiveBufferSize(
+                            Settings.ReceiveBufferSize ?? 256 * 1024)), backlog:4096);
+            var mappedAddr = _listenAddress is IPEndPoint p
+                ? p
+                : await DnsToIPEndpoint(_listenAddress as DnsEndPoint);
+                            
+            _addr = DnsHelpers.MapSocketToAddress(
+                socketAddress: mappedAddr, 
+                schemeIdentifier: SchemeIdentifier,
+                systemName: base.System.Name,
+                hostName: Settings.PublicHostname,
+                publicPort: Settings.PublicPort);
+            //_connectionSource.Via
+            _serverBindingTask = _connectionSource.Select
+            (ic =>
+            {
+                
+                AssociationListenerPromise.Task.ContinueWith(r =>
+                {
+                    var listener = r.Result;
+                    var remoteAddress =
+                        DnsHelpers.MapSocketToAddress(
+                            (IPEndPoint)ic.RemoteAddress, "tcp", base.System.Name);
+                    var queuePromise =
+                        new TaskCompletionSource<
+                            ISourceQueueWithComplete<IO.ByteString>>();
+                    StreamingTcpAssociationHandle handle;
+                    handle = new StreamingTcpAssociationHandle(_addr,
+                        remoteAddress,
+                        queuePromise);
+                    handle.ReadHandlerSource.Task.ContinueWith(s =>
+                    {   
+                        var otherListener = s.Result;
+                        handle.RegisterListener(otherListener);
+                    }, TaskContinuationOptions.ExecuteSynchronously);
+                    var outQueue = ic.HandleWith(buildFlow(handle)
+                        .Recover(
+                            ex =>
+                            {
+                                handle.Notify(
+                                    new Disassociated(DisassociateInfo.Unknown));
+                                return IO.ByteString.Empty;
+                            }), _mat);
+                    queuePromise.SetResult(outQueue);
+                    listener.Notify(new InboundAssociation(handle));
+                });
+                return NotUsed.Instance;
+            }).ToMaterialized(Sink.Ignore<NotUsed>(),Keep.Left).Run(_mat);
```

Comment:
We need to pass in some hints to the materializer here as to which dispatcher the materialized actors should run on, IIRC

Reply:
the /core/akka.remote/ code has dispatcher hints :)

Reply:
Ah ok - I was wondering if I was missing something